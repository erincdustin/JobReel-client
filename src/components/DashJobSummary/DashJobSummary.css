.DashJobSummary > h4 {
  text-align: center;
}

.options > ul > li {
  list-style: none;
  width: 100%;
}

.ul {
  width: 95%;
  margin: auto;
  height: 100%;
  background: #3b3a50;
  padding: 5px 10px;
  border-radius: 10px;
}

.viewAll {
  text-align: center;
  margin: 15px auto
}
