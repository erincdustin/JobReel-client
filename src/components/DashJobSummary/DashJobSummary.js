import React from 'react';
import { Link } from 'react-router-dom';
import './DashJobSummary.css';
import jobReelContext from '../../context/JobReelContext';
import jobReelApiService from '../../services/jobreel-api-service';

class DashJobSummary extends React.Component {

  static contextType = jobReelContext

  componentDidMount() {
    if (this.context.savedJobs.length === 0) {
      jobReelApiService.getSavedJobs()
        .then(res => {
          this.context.setSavedJobs(res);
        })
    }
  }

  renderJobSummaries = () => {
    let jobs = Array.from(this.context.savedJobs);
    jobs = jobs.sort((a, b) => {
      return new Date(b.date_added) - new Date(a.date_added);
    });
    if (this.props.section === "APPLIED JOBS") {
      jobs = jobs.filter(job => {
        return job.status === 'Applied';
      })
    } else {
      jobs = jobs.filter(job => {
        return job.status === 'Interested';
      })
    }
    jobs = jobs.slice(0, 3).map(job => {
      return (
        <li key={job.job_id}>{job.job_title}, {job.company} - <a href={job.url}>{job.url}</a></li>
      )
    })
    return (
      <ul>
        {jobs}
      </ul>
    )
  }
  
  render() {
    return (
      <>
        <div className="DashJobSummary">
          <h4>{this.props.section}</h4>
        </div>
        <div className='options'>
            {this.renderJobSummaries()}
        </div>
        <Link to="/saved-jobs">View all</Link>
      </>
    )
  }
}

export default DashJobSummary;