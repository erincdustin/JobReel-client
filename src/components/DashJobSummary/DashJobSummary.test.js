import React from 'react';
import ReactDOM from 'react-dom'
import { BrowserRouter } from 'react-router-dom'
import DashJobSummary from './DashJobSummary'

describe('DashJobSummary smoke test', () => {
  it('renders without crashing', () => {
      const div = document.createElement('div');
      ReactDOM.render(<BrowserRouter><DashJobSummary /></BrowserRouter>, div);
      ReactDOM.unmountComponentAtNode(div);
  })
})