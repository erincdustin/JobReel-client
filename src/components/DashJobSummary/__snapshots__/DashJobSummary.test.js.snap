// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DashJobSummary snapshot test renders the UI as expected 1`] = `<DashJobSummary />`;
